grails-app/domain/com/unifina/domain/marketplace/Product.groovy

```diff
@@ -29,7 +29,7 @@ class Product {
 	// The below fields exist in the domain object for speed & query support, but the ground truth is in the smart contract.
 	String ownerAddress
 	String beneficiaryAddress
-	Long pricePerSecond
+	Long pricePerSecond = 0
```

Comment:
Do we need default value here if we already have one in `CreateProductCommand`?

Reply:
@harbu I thought the same and actually tried to refactor it to be more DRY, but ended up rolling back and keeping the duplication, as it was already there, and other approaches were causing problems:

- ”Defaults only in domain object”: In the ProductService, currently the *command object* is being validated instead of the domain object. This approach requires the defaults to be in the command object. I tried to change it so that the command object props are applied to the Product first, and then *the Product* is validated, but ran into some weird GORM validation behavior which I couldn’t solve quickly.

- ”Defaults only in command object”: This would be another valid approach, but would break many tests which are doing just new Product(), and would make the default values harder to find in the code.

It could be refactored later but I decided to limit the scope of this PR to the small change I was making.